xonsh/shell.py

```diff
@@ -161,6 +161,8 @@ def push(self, line):
 
     def reset_buffer(self):
         """Resets the line buffer."""
+        cmd = ''.join(list(filter(('\n').__ne__, self.buffer)))  
```

Comment:
The cast to list after calling `filter` should be unnecessary (i.e., `str.join` can be called with a `filter` object as its argument).

Actually, what is the reason for the filtering?  I'm not sure I see it.  That is, couldn't we just store `''.join(self.buffer)`?